perses/samplers/samplers.py

```diff
@@ -1087,39 +1087,38 @@ def _ncmc_geometry_ncmc(self, topology_proposal, positions, old_log_weight, new_
         if self.verbose: print("Updating chemical state with ncmc-geometry-ncmc scheme...")
 
         initial_time = time.time()
-        ncmc_old_positions, ncmc_elimination_logp, potential_delete = self._ncmc_delete(topology_proposal, positions)
+        ncmc_old_positions, ncmc_elimination_logp = self._ncmc_delete(topology_proposal, positions)
 
         geometry_old_positions = ncmc_old_positions
         geometry_new_positions, geometry_logp_propose = self._geometry_forward(topology_proposal, geometry_old_positions)
 
         geometry_logp_reverse = self._geometry_reverse(topology_proposal, geometry_new_positions, geometry_old_positions)
-        geometry_logp = geometry_logp_reverse - geometry_logp_propose
-
-        ncmc_new_positions, ncmc_introduction_logp, potential_insert = self._ncmc_insert(topology_proposal, geometry_new_positions)
 
-        # Compute change in eliminated potential contribution.
-        switch_logp = - (potential_insert - potential_delete)
+        ncmc_new_positions, ncmc_introduction_logp = self._ncmc_insert(topology_proposal, geometry_new_positions)
 
         # Compute total log acceptance probability, including all components.
-        logp_accept = topology_proposal.logp_proposal + geometry_logp + switch_logp + ncmc_elimination_logp + ncmc_introduction_logp + new_log_weight - old_log_weight
+        logp_accept = (topology_proposal.logp_proposal + ncmc_elimination_logp + geometry_logp_propose + geometry_logp_reverse + ncmc_introduction_logp) + (new_log_weight - old_log_weight)
```

Comment:
Hm, what is the new component of the acceptance criterion here that you said was missing?